### ISSUE TITLE
typescript-eslint's "no any" rules don't work with inferred function types from sveltekit

### ISSUE THREAD

**Issue by DetachHead** [ota-meshi/eslint-plugin-svelte#413]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.33.0

### What version of `eslint-plugin-svelte` are you using?

2.22.0

### What did you do?

<details>
<summary>Configuration</summary>

```json
{
    "rules": {
        "@typescript-eslint/no-unsafe-member-access": "error"
    }
}
```
</details>

```ts
// routes/api/+server.ts
export const GET = async ({ request }) => {
    // request is not any, since it's inferred as Request by sveltekit
    request.headers // error: @typescript-eslint/no-unsafe-member-access
}
```

### What did you expect to happen?

no error

### What actually happened?

`@typescript-eslint/no-unsafe-member-access` error

### Link to **GitHub Repo** with Minimal Reproducible Example

i can make a github repo for this later if you like but since you're already aware of the underlying issue (#390) i assume this isn't needed

### Additional comments

originally raised at https://github.com/ota-meshi/eslint-plugin-svelte/issues/390#issuecomment-1477369726

**Comment by ota-meshi** [comment]
Where are the types for `GET` sourced from? I don't understand the code you provided because there are no type declarations. Could you please share a repository to reproduce the issue?
I've only made static sites using svelte kit, so I'm not sure :sweat:

**Comment by DetachHead** [comment]
sveltekit infers the types so you don't have to annotate them explicitly. i've updated the issue with a github repo. to reproduce the error, checkout that branch then run the following commands
```
npm install
npm run check
eslint .
```

**Comment by polyzen** [comment]
Think you mean `npx eslint .`

**Comment by DetachHead** [comment]
true. i added `.\node_modules\.bin` to my path so i forgot about `npx`. updated my comment

**Comment by polyzen** [comment]
Inferring types is a new feature: https://svelte.dev/blog/zero-config-type-safety

**Comment by ota-meshi** [comment]
memo: Perhaps the following source code is injecting the type.

https://github.com/sveltejs/language-tools/blob/ac827294b4aff00ecac961b164155f29c0f5ed1d/packages/svelte2tsx/src/helpers/sveltekit.ts#L189

**Comment by ota-meshi** [comment]
Hmm.. We may need new typescript parser for the eslint (not svelte parser). We should check if it integrates well with `svelte2tsx`.

**Comment by jasonpanosso** [comment]
I still have this issue, is there any progress on this, or any known workaround/solution?

**Comment by thenbe** [comment]
Another minimal repro here: https://github.com/thenbe/repro-eslint-svelte-typecheck

Eslint is not able to

### Reproduction steps

1. Clone and `pnpm install`
1. Run: `pnpm eslint src/routes/+page.svelte`

Expected: We should get no eslint errors.
Actual: We get two eslint errors. Error message:

```
src/routes/+page.svelte
   7:20  error  Unsafe member access .url on an `any` value       @typescript-eslint/no-unsafe-member-access
  11:8   error  Unsafe member access .greeting on an `any` value  @typescript-eslint/no-unsafe-member-access
```

### Other

`svelte-check` (i.e. `pnpm run check`) passes without errors. Only eslint shows error.

**Comment by ryoppippi** [comment]
Got the same issue....

**Comment by keve1227** [comment]
A workaround I found is to import the offending type(s) explicitly:

```ts
import type { RequestEvent } from "./$types";

export async function GET({ url }: RequestEvent) {
  return new Response(url.searchParams.get("message"));
}
```

**Comment by thenbe** [comment]
> A workaround is to import the offending type explicitly

I wasn't sure how to apply this workaround to the minimal repro posted in the [comment](https://github.com/sveltejs/eslint-plugin-svelte/issues/413#issuecomment-2171792878) above.

The issue that remains is that eslint fails to see the inferred types not only in the [`+page.ts`](https://github.com/thenbe/repro-eslint-svelte-typecheck/blob/main/src/routes/%2Bpage.ts) file, but in its sibling [`+page.svelte`](https://github.com/thenbe/repro-eslint-svelte-typecheck/blob/main/src/routes/%2Bpage.svelte) as well.
